lms/static/scripts/frontend_apps/components/LMSFilePicker.js

```diff
@@ -318,20 +319,33 @@ export default function LMSFilePicker({
   const { continueButton, warningOrError } =
     options[dialogState.continueAction];
 
-  if (dialogState.state === 'fetching') {
-    return null;
+  // During the first fetch and load, we don't know yet whether we'll have
+  // success fetching files, or whether we'll need auth'z or hit another
+  // error. To avoid hopscotching between modals, show only
+  // a full-screen spinner during this state.
+  const isLoading = dialogState.state === 'fetching';
+  // During subsequent file/folder fetches, represent the loading state within
+  // the file list UI (do not show a full-screen spinner)
+  const isReloading = dialogState.state === 'reloading';
+  // Render the file list and breadcrumbs UI in certain states. This also
+  // requires more real estate in the <Modal>, and applies an appropriate CSS
+  // class
+  const withFileUI = ['reloading', 'fetched'].includes(dialogState.state);
+
+  if (isLoading) {
+    return <FullScreenSpinner />;
   }
 
   return (
-    <Dialog
-      contentClass="LMSFilePicker__dialog"
+    <Modal
+      contentClass={withFileUI ? 'LMSFilePicker' : ''}
```

Comment:
I've already got an issue tracking that `Modal` (and `Dialog`) use a legacy naming on this prop; it should be `classes`: https://github.com/hypothesis/frontend-shared/issues/181